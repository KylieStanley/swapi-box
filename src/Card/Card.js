import React from 'react'
import './Card.scss'
import PropTypes from 'prop-types'
import Images from '../images'
import Side from '../side'

const Card = (props) => {
  const allInfo = Object.keys(props.item).filter((key) => {
    return key !== 'favorite'
  })

  const info = allInfo.map((key) => {
    if (key === 'name') {
      return <h1 key={key}>{props.item[key]}</h1>
    } else {
      return <p key={key}>{key}: {props.item[key]}</p>
    }
  })

  const favorited = !props.item.favorite ? 'white-saber' : Side[props.item.name] === 'light' ? 'green-saber' : 'red-saber'
  return (
    <div className="card">
      <div className="card-info">
        { info }
      </div>
      <img className="card-image" src={Images[props.item.name]} alt={Images[props.item.name]} />
      <div className="gradient"></div>
      <button className="favorite" onClick={() => props.handleFavorite(props.item, props.itemType)}>
        <img src={Images[favorited]} className="white-saber" alt="lightsaber" />
      </button>
      <div className="card-text-background"></div>
    </div>
  )
}

export default Card

Card.propTypes = {
  item: PropTypes.object.isRequired,
  handleFavorite: PropTypes.func.isRequired,
  itemType: PropTypes.string.isRequired
}
