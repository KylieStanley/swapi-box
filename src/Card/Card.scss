$main-blue: #03b9ffb0;

.card {
  align-items: center;
  border-radius: 4px;
  color: #fff;
  display: flex;
  flex-wrap: wrap;
  height: 300px;
  justify-content: center;
  margin: 12px;
  text-align: center;
  position: relative;
  width: 300px;

  &::before,
  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
  }

  &:hover::after {
    border-top: 2px solid #0081ba99;
    transform: rotate(270deg);
    transition:
      transform .6s linear 0s,
      border-left-width 0s linear 0.35s; 
  }

  &:hover {
    box-shadow: 0 0 24px 1px $main-blue, 0 0 10px 1px #03b9ffd1;
    transition: 2s;
  }

  &:hover .card-info {
    display: block;
    opacity: 1;
  }

  &:hover .card-text {
    display: none;
  }

  &:hover .card-image {
    opacity: .4;
    filter: blur(2px);
    transition: .3s;
  }
}

.card-image {
  width:100%;
  height:100%;
/*  border-radius: 50%;*/
  position: absolute;
  filter: drop-shadow(5px 5px 5px #333) drop-shadow(-5px -5px 5px #333);
}

.card-text {
  bottom: 0px;
  width: 100%;
  position: absolute;
  z-index: 2;
}

.card-info {
  opacity: 0;
  z-index: 2;
  transition: .5s;
  width: 200px;

  &:before,
  &:after {
    content: '';
    position: absolute;
    z-index: 1;
  }
}

.favorite {
  position: absolute;
  z-index: 2;
  top: 5px;
  right: 5px;
  background: none;
  border: none;
}

.white-saber {
  height: 55px;
  transform: scaleX(-1);
  filter: FlipH;
  filter: drop-shadow(2px 2px 5px #333);
}

.white-saber:hover {
animation: grow;
animation-duration: .5s; 
}

.white-saber { transition: all .2s ease-in-out; }
.white-saber:hover { transform: scale(1.5); }