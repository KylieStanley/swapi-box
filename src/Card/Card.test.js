import React from 'react'
import { shallow, mount } from 'enzyme'
import Card from './Card.js'

describe('Card', () => {
  it('should match the snapshot', () => {
    const mockPerson = {
      'name': 'Luke Skywalker',
      'homeworld': 'Tatooine',
      'species': 'human',
      'population': 20000000,
      'favorite': false
    }
    const wrapper = shallow(<Card person={ mockPerson }/>)


    expect(wrapper).toMatchSnapshot()
  })
})