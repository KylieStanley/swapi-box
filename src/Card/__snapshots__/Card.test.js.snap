// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    handleFavorite={[MockFunction]}
    item={
      Object {
        "favorite": false,
        "homeworld": "Tatooine",
        "name": "Luke Skywalker",
        "population": 20000000,
        "species": "human",
      }
    }
    itemType="people"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-info"
        >
          <h1>
            Luke Skywalker
          </h1>
          <p>
            homeworld
            : 
            Tatooine
          </p>
          <p>
            species
            : 
            human
          </p>
          <p>
            population
            : 
            20000000
          </p>
        </div>,
        <img
          alt="../images/Luke.png"
          className="card-image"
          src="../images/Luke.png"
        />,
        <div
          className="gradient"
        />,
        <button
          className="favorite"
          onClick={[Function]}
        >
          <img
            alt="lightsaber"
            className="white-saber"
            src="../images/white-lightsaber.svg"
          />
        </button>,
        <div
          className="card-text-background"
        />,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Luke Skywalker
            </h1>,
            <p>
              homeworld
              : 
              Tatooine
            </p>,
            <p>
              species
              : 
              human
            </p>,
            <p>
              population
              : 
              20000000
            </p>,
          ],
          "className": "card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "name",
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": "homeworld",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "homeworld",
                ": ",
                "Tatooine",
              ],
            },
            "ref": null,
            "rendered": Array [
              "homeworld",
              ": ",
              "Tatooine",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "species",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "species",
                ": ",
                "human",
              ],
            },
            "ref": null,
            "rendered": Array [
              "species",
              ": ",
              "human",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "population",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population",
                ": ",
                20000000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "population",
              ": ",
              20000000,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "../images/Luke.png",
          "className": "card-image",
          "src": "../images/Luke.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "gradient",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="lightsaber"
            className="white-saber"
            src="../images/white-lightsaber.svg"
          />,
          "className": "favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "lightsaber",
            "className": "white-saber",
            "src": "../images/white-lightsaber.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "card-text-background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-info"
          >
            <h1>
              Luke Skywalker
            </h1>
            <p>
              homeworld
              : 
              Tatooine
            </p>
            <p>
              species
              : 
              human
            </p>
            <p>
              population
              : 
              20000000
            </p>
          </div>,
          <img
            alt="../images/Luke.png"
            className="card-image"
            src="../images/Luke.png"
          />,
          <div
            className="gradient"
          />,
          <button
            className="favorite"
            onClick={[Function]}
          >
            <img
              alt="lightsaber"
              className="white-saber"
              src="../images/white-lightsaber.svg"
            />
          </button>,
          <div
            className="card-text-background"
          />,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Luke Skywalker
              </h1>,
              <p>
                homeworld
                : 
                Tatooine
              </p>,
              <p>
                species
                : 
                human
              </p>,
              <p>
                population
                : 
                20000000
              </p>,
            ],
            "className": "card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "name",
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker",
              },
              "ref": null,
              "rendered": "Luke Skywalker",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": "homeworld",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "homeworld",
                  ": ",
                  "Tatooine",
                ],
              },
              "ref": null,
              "rendered": Array [
                "homeworld",
                ": ",
                "Tatooine",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "species",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "species",
                  ": ",
                  "human",
                ],
              },
              "ref": null,
              "rendered": Array [
                "species",
                ": ",
                "human",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "population",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population",
                  ": ",
                  20000000,
                ],
              },
              "ref": null,
              "rendered": Array [
                "population",
                ": ",
                20000000,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "../images/Luke.png",
            "className": "card-image",
            "src": "../images/Luke.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "gradient",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="lightsaber"
              className="white-saber"
              src="../images/white-lightsaber.svg"
            />,
            "className": "favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "lightsaber",
              "className": "white-saber",
              "src": "../images/white-lightsaber.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "card-text-background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
